src/LCRmeter-Keithley_590/main.py

```diff
@@ -0,0 +1,335 @@
+        # Currently only used mode is (One-shot, talk) because it works
+        self.port.write("T0,0X")
+
+        # Read out buffer
+        self.port.write("B0X")
+        answer = self.port.read()
+
+        # Answer has format ZTSK 1.23E+0, 4.56E+0, 7.89E+0, values are C, G, V
+        answer = answer.split(",")
+
+        self.capacitance = float(answer[0][4:])  # Remove prefix of type ZTPK
+        self.conductance = float(answer[1])
+        self.measured_dc_bias = float(answer[2])
+
+        # TODO: Could convert to resistance and reactance for post-processing in module
+        # resistance = 1 / self.conductance
+        # frequency = 1E8 if self.frequency == "100kHz" else 1E9
+        # reactance = 1 / (2 * np.pi * frequency * self.capacitance)
+
+    def call(self) -> tuple:
+        """Return measured values."""
+        return self.capacitance, self.conductance, self.measured_dc_bias
+
+    """ Wrapped functions - See Manual 4-19 for more commands """
+
+    def set_frequency(self, frequency: str) -> None:
+        """Set the measurement frequency."""
+        command = "F0X" if frequency == "100kHz" else "F1X"
+        self.port.write(command)
+
+    def set_range(self, measurement_range: str) -> None:
+        """Set the measurement range."""
+        if self.frequency == "100kHz":
+            # TODO: Add 10x amplifier boolean to GUI
+            ranges = {
+                "Auto": "R0",
+                "2pF/2uS": "R1",
+                "20pF/20uS": "R2",
+                "200pF/200uS": "R3",
+                "2nF": "R4",
+                "R1 x 10": "R5",
+                "R2 x 10": "R6",
+                "R3 x 10": "R7",
+                "R4 x 10": "R8",
+                "Auto off": "R9",
+            }
+        else:
+            ranges = {
+                "Auto": "R0",
+                "20pF/200uS": "R1",
+                # "20pF/200uS": "R2",  # doubled?
+                "200pF/2mS": "R3",
+                "2nF/20mS": "R4",
+                "Auto off": "R9",
+            }
+
+        try:
+            range_command = ranges[measurement_range]
+        except KeyError as e:
+            msg = f"Range {measurement_range} not available for frequency {self.frequency}."
+            range_command = ranges["Auto"]
+            raise ValueError(msg) from e
+
+        self.port.write(f"{range_command}X")
+
+    def set_reading_rate(self, rate: str) -> None:
+        """Set the reading rate."""
+        command = f"S{self.reading_rates[rate]}X"
+        self.port.write(command)
+
+    def set_trigger_mode_and_source(self, trigger: str) -> None:
+        """Set the trigger mode and source."""
+        command = f"T{self.trigger_modes[trigger]}X"
+        self.port.write(command)
+
+    def set_zero_on(self, set_on: bool = True) -> None:
+        """Set the zero."""
+        command = "Z1X" if set_on else "Z0X"
+        self.port.write(command)
```

Comment:
What does it mean to "set zero"? And it is not required to set it off again after the measurement / unconfigure?